File: Android/sources/androidx/emoji2/widget/EmojiExtractEditText.java
package androidx.emoji2.widget;

import android.content.Context;
import android.content.res.TypedArray;
import android.inputmethodservice.ExtractEditText;
import android.text.method.KeyListener;
import android.text.method.NumberKeyListener;
import android.util.AttributeSet;
import android.view.ActionMode;
import android.view.inputmethod.EditorInfo;
import android.view.inputmethod.InputConnection;
import kotlin.jvm.internal.IntCompanionObject;

public class EmojiExtractEditText extends ExtractEditText {
    public y15 a;
    public final boolean b;

    public EmojiExtractEditText(Context context, AttributeSet attributeSet) {
        super(context, attributeSet);
        if (!this.b) {
            this.b = true;
            int i = 0;
            if (attributeSet != null) {
                TypedArray obtainStyledAttributes = getContext().obtainStyledAttributes(attributeSet, goc.EmojiEditText, 16842862, 0);
                int integer = obtainStyledAttributes.getInteger(goc.EmojiEditText_maxEmojiCount, IntCompanionObject.MAX_VALUE);
                obtainStyledAttributes.recycle();
                i = integer;
            }
            setMaxEmojiCount(i);
            setKeyListener(super.getKeyListener());
        }
    }

    private y15 getEmojiEditTextHelper() {
        if (this.a == null) {
            this.a = new y15(this, true);
        }
        return this.a;
    }

    public int getEmojiReplaceStrategy() {
        return getEmojiEditTextHelper().c;
    }

    public int getMaxEmojiCount() {
        return getEmojiEditTextHelper().b;
    }

    public final InputConnection onCreateInputConnection(EditorInfo editorInfo) {
        return getEmojiEditTextHelper().a(super.onCreateInputConnection(editorInfo), editorInfo);
    }

    public void setCustomSelectionActionModeCallback(ActionMode.Callback callback) {
        super.setCustomSelectionActionModeCallback(callback);
    }

    public void setEmojiReplaceStrategy(int i) {
        y15 emojiEditTextHelper = getEmojiEditTextHelper();
        emojiEditTextHelper.c = i;
        ((w35) emojiEditTextHelper.a.b).v = i;
    }

    public void setKeyListener(KeyListener keyListener) {
        if (keyListener != null) {
            getEmojiEditTextHelper().a.getClass();
            if (!(keyListener instanceof j25) && !(keyListener instanceof NumberKeyListener)) {
                keyListener = new j25(keyListener);
            }
        }
        super.setKeyListener(keyListener);
    }

    public void setMaxEmojiCount(int i) {
        y15 emojiEditTextHelper = getEmojiEditTextHelper();
        emojiEditTextHelper.getClass();
        if (i >= 0) {
            emojiEditTextHelper.b = i;
            ((w35) emojiEditTextHelper.a.b).o = i;
            return;
        }
        throw new IllegalArgumentException("maxEmojiCount should be greater than 0");
    }
}

File: IOS/v2/iosx/emoji2/widget/EmojiExtractEditText.swift
package iOSx.emoji2.widget;.swift
import iOS.content.Context;.swift
import iOS.content.res.TypedArray;.swift
import iOS.inputmethodservice.ExtractEditText;.swift
import iOS.text.method.KeyListener;.swift
import iOS.text.method.NumberKeyListener;.swift
import iOS.util.AttributeSet;.swift
import iOS.view.ActionMode;.swift
import iOS.view.inputmethod.EditorInfo;.swift
import iOS.view.inputmethod.InputConnection;.swift
import kotlin.jvm.internal.IntCompanionObject;.swift
public class EmojiExtractEditText extends ExtractEditText {.swift
public y15 a;.swift
public final boolean b;.swift
public EmojiExtractEditText(Context context, AttributeSet attributeSet) {.swift
super(context, attributeSet);.swift
if (this.b) {.swift
this.b = true;.swift
int i = 0;.swift
if (attributeSet = null) {.swift
TypedArray obtainStyledAttributes = getContext().obtainStyledAttributes(attributeSet, goc.EmojiEditText, 16842862, 0);.swift
int integer = obtainStyledAttributes.getInteger(goc.EmojiEditText_maxEmojiCount, IntCompanionObject.MAX_VALUE);.swift
obtainStyledAttributes.recycle();.swift
i = integer;.swift
}.swift
setMaxEmojiCount(i);.swift
setKeyListener(super.getKeyListener());.swift
}.swift
}.swift
private y15 getEmojiEditTextHelper() {.swift
if (this.a == null) {.swift
this.a = new y15(this, true);.swift
}.swift
return this.a;.swift
}.swift
public int getEmojiReplaceStrategy() {.swift
return getEmojiEditTextHelper().c;.swift
}.swift
public int getMaxEmojiCount() {.swift
return getEmojiEditTextHelper().b;.swift
}.swift
public final InputConnection onCreateInputConnection(EditorInfo editorInfo) {.swift
return getEmojiEditTextHelper().a(super.onCreateInputConnection(editorInfo), editorInfo);.swift
}.swift
public void setCustomSelectionActionModeCallback(ActionMode.Callback callback) {.swift
super.setCustomSelectionActionModeCallback(callback);.swift
}.swift
public void setEmojiReplaceStrategy(int i) {.swift
y15 emojiEditTextHelper = getEmojiEditTextHelper();.swift
emojiEditTextHelper.c = i;.swift
((w35) emojiEditTextHelper.a.b).v = i;.swift
}.swift
public void setKeyListener(KeyListener keyListener) {.swift
if (keyListener = null) {.swift
getEmojiEditTextHelper().a.getClass();.swift
if ((keyListener instanceof NumberKeyListener)) {.swift
keyListener = new j25(keyListener);.swift
}.swift
}.swift
super.setKeyListener(keyListener);.swift
}.swift
public void setMaxEmojiCount(int i) {.swift
y15 emojiEditTextHelper = getEmojiEditTextHelper();.swift
emojiEditTextHelper.getClass();.swift
if (i >= 0) {.swift
emojiEditTextHelper.b = i;.swift
((w35) emojiEditTextHelper.a.b).o = i;.swift
return;.swift
}.swift
throw new IllegalArgumentException("maxEmojiCount should be greater than 0");.swift
}.swift
}.swift
